platform/mime/core/platform-mime-core-impl/src/test/java/ddf/mime/mapper/MimeTypeMapperTest.java

```diff
@@ -123,4 +132,48 @@ public void testGuessMimeTypeForXmlIngest() throws Exception {
         assertNull(mimeType);
     }
 
+    @Test
+    public void testGuessMimeTypeForXmlIngestNoExtension() throws Exception {
+        MimeTypeMapper mapper = new MimeTypeMapperImpl(MOCK_MIME_TYPE_RESOLVERS);
+
+        InputStream is = FileUtils.openInputStream(new File(CSW_RECORD_FILE_NO_EXTENSION));
+        String mimeType = mapper.guessMimeType(is, "");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("text/xml;id=csw", mimeType);
+
+        is = FileUtils.openInputStream(new File(XML_METACARD_FILE_NO_EXTENSION));
+        mimeType = mapper.guessMimeType(is, "");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("text/xml", mimeType);
+    }
+
+    @Test
+    public void testGuessMimeTypeForFileExtension() throws Exception {
+        MimeTypeMapper mapper = new MimeTypeMapperImpl(MOCK_MIME_TYPE_RESOLVERS);
+
+        String mimeType = mapper.getMimeTypeForFileExtension("xml");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("text/xml;id=csw", mimeType);
+
+        mimeType = mapper.getMimeTypeForFileExtension(".nitf");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("image/nitf", mimeType);
+
+        //change the priority order to get a different resolver for .xml
+        List<MimeTypeResolver> resolversWithXmlPriority = new ArrayList<>();
+        resolversWithXmlPriority.add(new MockMimeTypeResolver("XmlMetacardResolver",
+                15,
+                new String[] {"xml=text/xml"},
+                "urn:catalog:metacard"));
+        resolversWithXmlPriority.add(new MockMimeTypeResolver("CswResolver",
+                10,
+                new String[] {"xml=text/xml;id=csw"},
+                "http://www.opengis.net/cat/csw/2.0.2"));
+
+        mapper = new MimeTypeMapperImpl(resolversWithXmlPriority);
+        mimeType = mapper.getMimeTypeForFileExtension("xml");
+        LOGGER.debug("mimeType = {}", mimeType);
+        assertEquals("text/xml", mimeType);
```

Comment:
The second part isn't really necessary. Was just demonstrating that it doesn't bound it to the text/xml resolver. I'm going to remove that part though